collect_app/src/main/java/org/odk/collect/android/storage/migration/StorageMigrationRepository.java

```diff
@@ -0,0 +1,34 @@
+package org.odk.collect.android.storage.migration;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+public class StorageMigrationRepository {
+    private MutableLiveData<StorageMigrationResult> migrationResult = new MutableLiveData<>();
+
+    private boolean isMigrationBeingPerformed;
+
+    public LiveData<StorageMigrationResult> getResult() {
+        return migrationResult;
+    }
+
+    public void setResult(StorageMigrationResult storageMigrationResult) {
+        migrationResult.postValue(storageMigrationResult);
+    }
+
+    public boolean isMigrationBeingPerformed() {
+        return isMigrationBeingPerformed;
+    }
+
+    void markMigrationStart() {
+        isMigrationBeingPerformed = true;
+    }
+
+    void markMigrationEnd() {
+        isMigrationBeingPerformed = false;
+    }
+
+    public void consumeResult() {
```

Comment:
Now I use it in `MainMenuActivity.onDestroy()` otherwise if you close the app and open it again the banner with success message will be visible because the status persist. We agreed that the banner in such a case should be gone.